src/main.c

```diff
@@ -1495,6 +1495,8 @@ static void fork_server(int fd, conn_t *l, unsigned char no_fork) {
     pr_session_disconnect(NULL, PR_SESS_DISCONNECT_SESSION_INIT_FAILED, NULL);
   }
 
+  pr_event_generate("core.connected", conn);
```

Comment:
We generate a new `core.connected` event only if all modules have successfully initialized themselves for this connection.

In the future, the following debug logging can be moved into an event listener for this event.